ql/src/java/org/apache/hadoop/hive/ql/cache/results/QueryResultsCache.java

```diff
@@ -260,6 +252,7 @@ private CacheEntryStatus setStatus(CacheEntryStatus newStatus) {
       synchronized (this) {
         CacheEntryStatus oldStatus = status;
         status = newStatus;
+        this.notifyAll();
         return oldStatus;
```

Comment:
Notify waiting threads when the status changes.  This is the correct way to do this instead of putting `notifyAll` in a bunch of different places.